### PR TITLE
 CTCTRADERS-266: redirection logic

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/connector/DestinationConnector.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DestinationConnector @Inject()(val config: AppConfig, val http: HttpClient)(implicit ec: ExecutionContext) {
+
+  def sendMessage(responseData: String, headers: Seq[(String, String)])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
+
+    val serviceUrl = s"${config.traderAtDestinationUrl}/message"
```
Comment (chris-ianson) [inline app/connector/DestinationConnector.scala:30]: Could the _message_ be part of the traderAtDestinationUrl?

File: conf/application.conf [hunk 2]
```diff
@@ -162,6 +162,13 @@ microservice {
       host = localhost
       port = 8500
     }
+
+    destination {
```
Comment (chris-ianson) [inline conf/application.conf]: Could we maybe rename this to transit-movements-trader-at-destination? Just so it's clear where it's going (_destination_ could be generic)

File: test/controllers/MessageControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  private val application = applicationBuilder.overrides(bind[DestinationConnector].toInstance(mockConnector)).build()
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(mockConnector)
+  }
+
+  "MessageController must" - {
+    "return ok for the message Type Goods release" in {
+      when(mockConnector.sendMessage(any(), any())(any(), any())).thenReturn(Future.successful(HttpResponse(OK)))
```
Comment (chris-ianson) [inline test/controllers/MessageControllerSpec.scala:43]: Could we include the _X-Message-Type_ header/value to ensure it's being passed along and used in the sendMessage method?

File: app/controllers/MessageType.scala [hunk 4]
```diff
@@ -14,20 +14,10 @@
...
  */
 
-package uk.gov.hmrc.transitmovementstraderrouter.controllers
+package controllers
 
-import javax.inject.{Inject, Singleton}
-import play.api.mvc.{Action, AnyContent, ControllerComponents}
-import uk.gov.hmrc.play.bootstrap.controller.BackendController
-import uk.gov.hmrc.transitmovementstraderrouter.config.AppConfig
+object MessageType {
```
Comment (chris-ianson) [inline app/controllers/MessageType.scala]: Consider moving this into a different package (is it relevant in controllers?)

File: app/connector/DestinationConnector.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply app/connector/DestinationConnector.scala:30]: @chris-ianson later in future if we need to call some other end point in trader-at-destination
for flexibility better to keep traderAtDestinationUrl as it is rather then binding specific to message url

File: conf/application.conf (hunk 2 above)
Comment (ashwinirudrappa) [reply conf/application.conf]: Chris can be done as we followed the same naming convention as frontend, we can discuss

File: app/connector/DestinationConnector.scala (hunk 1 above)
Comment (chris-ianson) [reply app/connector/DestinationConnector.scala:30]: serviceUrl is only used in one place. The url should include message and if we need to change it in future we only have to change it in AppConfig

**Pull request merged**